### ISSUE TITLE
conditional debug

### ISSUE THREAD

**Issue by Haroenv** [preactjs/preact#3894]
**Describe the feature you'd love to see**

I'd like to do something like `if (myDevCondition) initDebug();`, so I can have different output files where that condition is either true or false. I know I could do `require('preact/debug')`, but that wouldn't translate to pure esm, as it would mean I'd need to process the `if` out of the code, which currently isn't yet possible.

Maybe `initDebug` from `preact/debug/src/debug.js` could be exposed somehow?

**Additional context (optional)**

In my libraries I'm using preact as a dependency, so people don't necessarily import it themselves.

**Comment by rschristian** [comment]
If you're happy to have multiple output files, why not configure your build tool (assuming you're using one) to inject `import 'preact/debug';` into some/one of your output files?

Edit: Exposing `initDebug` wouldn't help here as debug needs to run _before_ all other Preact imports as we rely on side effects & attaching to a global `options` instance.

**Comment by Haroenv** [comment]
Yes, that could indeed be one of the solutions. Most of our implementations indeed have multiple output files, especially in UMD, however the ESM version usually only has one file with conditionals à la `process.env.NODE_ENV`, for which I'm not really sure what the solution would be.

Thanks for your idea already!
